parsec_platform_interface/lib/method_channel_parsec.dart

```diff
@@ -1,16 +1,22 @@
 import 'package:flutter/services.dart';
 import 'parsec_platform.dart';
 
-const _channel = MethodChannel('parsec_flutter');
+const _channelName = 'parsec_flutter';
+const _evalMethodName = 'nativeEval';
 
-/// An implementation of [ParsecPlatformInterfacePlatform] that uses method channels.
+/// Method channel implementation for native platform communication.
+/// 
+/// Handles equation evaluation on Android, iOS, Windows, Linux, and macOS
+/// through platform-specific implementations using method channels.
 class MethodChannelParsec extends ParsecPlatform {
-  /// The method channel used to interact with the native platform.
+  final MethodChannel _methodChannel = const MethodChannel(_channelName);
+
   @override
-  Future<dynamic> nativeEval(String equation) {
-    return _channel.invokeMethod(
-      'nativeEval',
-      {'equation': equation},
-    ).then((result) => result);
+  Future<dynamic> nativeEval(String equation) async {
+    if (equation.trim().isEmpty) {
+      throw ArgumentError.value(equation, 'equation', 'Equation cannot be empty');
+    }
+
+    return _methodChannel.invokeMethod(_evalMethodName, {'equation': equation});
```

Comment:
Is this change necessary for `parsec_web` to work? 
If not, it can be done in a separate PR, so we don't have to update the `parsec_platform_interface` versions in all the `pubspec` files.

Reply:
@rodiguif did you comment in the correct code here?

Or you mean this one?

<img width="796" height="184" alt="image" src="https://github.com/user-attachments/assets/a044dbf3-a6b7-4e93-a622-2e5b48177a99" />

Reply:
- The changes to `parsec_platform_interface` bumped it from 0.1.x to 0.2.0, and the web implementation requires these changes. Since these are backward-compatible improvements (not breaking changes), and the web implementation actually needs these improvements to handle edge cases properly, I think these platform interface changes should actually stay with the web PR.

The improved parsing logic handles edge cases that are particularly important for web/JavaScript interop where data types might be handled differently. Let me revert my changes and keep the platform interface improvements with the web PR:

@rodiguif

Reply:
Please check here: https://github.com/oxeanbits/parsec_flutter/pull/28#discussion_r2328384263